scripts/smart-validate.sh

```
@@ -0,0 +1,223 @@
+
+        if [ -n "$TS_ERRORS" ]; then
+            echo "❌ TypeScript errors found:"
+            echo "$TS_ERRORS" | sed 's/^/     /'
+            exit 1
+        else
+            echo "  ✅ TypeScript OK"
+        fi
+
+        echo "  2️⃣  JSON syntax..."
+        JSON_ERRORS=""
+        for json_file in $(find src/data -name "*.json" 2>/dev/null); do
```

Shellcheck (0.11.0)

[warning] For loops over find output are fragile. Use find -exec or a while read loop.

(SC2044)